File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        string s = to_string(n);
        sort(s.begin() , s.end());

        int len = s.length();
        int d1 = s[len-1] -'0';
        int d2 = s[len -2 ] -'0';

        return d1 * d2;
    }
};

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        m1 = m2 = 0

        for ch in str(n):
            d = int(ch)
            if d > m1:
                m2 = m1
                m1 = d
            elif d > m2:
                m2 = d
        
        return m1 *m2